inference/models/florence2/florence2.py

```diff
@@ -32,6 +34,33 @@ def prepare_generation_params(
             "pixel_values": preprocessed_inputs["pixel_values"],
         }
 
+    def predict(self, image_in: Image.Image, prompt="", history=None, **kwargs):
+        model_inputs = self.processor(
+            text=prompt, images=image_in, return_tensors="pt"
+        ).to(self.model.device)
+        input_len = model_inputs["input_ids"].shape[-1]
+
+        with torch.inference_mode():
+            prepared_inputs = self.prepare_generation_params(
+                preprocessed_inputs=model_inputs
+            )
+            generation = self.model.generate(
+                **prepared_inputs,
+                max_new_tokens=1024,
+                do_sample=False,
+                early_stopping=False,
+                num_beams=3
+            )
+            generation = generation[0]
+            if self.generation_includes_input:
+                generation = generation[input_len:]
+            decoded = self.processor.decode(generation, skip_special_tokens=False)
+            parsed_answer = self.processor.post_process_generation(
+                decoded, task=prompt.split(">")[0] + ">", image_size=image_in.size
+            )
+
+        return (json.dumps(parsed_answer),)
```

Comment:
can you explain the changes you made to the predict function inherited from the transformers base class here?

Reply:
I changed the flags a little, max output tokens, num_beans, turning off early stopping, to include special tokens, and used the output parser from florence to turn the output into usable json. Before using the florence 2 settings i was getting gibberish text out of decoder.

For the json string, the model returned a string response before (the straight decoded tokens) so I dont see how this would pose an issue but I agree there might be a better solution. The issue I faced was something in the response stack is only allowing text responses, I couldnt track down what though.